### PR TITLE
fix(DB/Zone):improved Terokkar Forest

### PR DESCRIPTION
**Pull request by Knindzagxg**
## Changes Proposed:
- Missing SAI creature updates.
- Improvements to existing if possible.

## Tests Performed:
- Applied SQL without problems.
- Tested and changed few in Skettis area cause there was problem with overlap in their cast time.

## How to Test:
- Apply SQL(s).
- Restart Server (You can .reload creature_template for each creature as well, but will take time)
- Go to zone (Check PR name) and check creatures around - You can also spawn (.npc add) each by command.

## Source for Changes:
- [Wowhead](https://wowhead.com) Database and or comments.
- [WotlkDB](https://wotlkdb.com/) Multi checking trough Private DB's.
- [WowWiki](https://wowwiki-archive.fandom.com/wiki/Portal:Main) Can be useful for extra informations.
- [WowPedia](https://wow.gamepedia.com/Wowpedia) Cross checking trough comments if possible.
- [Retail](https://worldofwarcraft.com/en-gb/) In-game checking if previous don't have any required informations.

## Target Branch:
- [x] Master

## Operating System:
- Linux (Debian 10)

## How to Test AzerothCore PR's
When a PR is ready to be tested, it will be marked as **[Waiting to be tested]**.
You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:
http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1616252599323979000.sql [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+(21644, 0, 0, 0, 0, 0, 100, 0, 1000, 3000, 30000, 31000, 11, 13730, 32, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Skettis Wing Guard - In Combat - Cast \'13730\''),
+(21644, 0, 1, 0, 0, 0, 100, 0, 2500, 5000, 20000, 25000, 11, 35321, 33, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 'Skettis Wing Guard - In Combat - Cast \'35321\''),
+(21644, 0, 2, 0, 2, 0, 100, 1, 20, 80, 0, 0, 11, 40546, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Skettis Wing Guard - Between 20-80% Health - Cast \'40546\' (No Repeat)'),
+(21644, 0, 3, 4, 2, 0, 100, 1, 5, 30, 0, 0, 11, 8599, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Skettis Wing Guard - Between 5-30% Health - Cast \'8599\' (No Repeat)'),
+(21644, 0, 4, 0, 61, 0, 100, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Skettis Wing Guard - Between 5-30% Health - Say Line 0 (No Repeat)');
```
Comment (pklloveyou) [inline data/sql/updates/pending_db_world/rev_1616252599323979000.sql:23]: It already has AI

Comment (pklloveyou) [reply data/sql/updates/pending_db_world/rev_1616252599323979000.sql:23]: I don't know if it's better than it is now

**Comment by pklloveyou** [comment]
First of all, it's better to have AI than none. I'll list individual creatures that have AI.

File: data/sql/updates/pending_db_world/rev_1616252599323979000.sql (hunk 1 above)
Comment (Knindzagxg) [reply data/sql/updates/pending_db_world/rev_1616252599323979000.sql:23]: Yeah, those i updated had kinda bad AI.
Overlaping with spells mostly.

**Comment by Helias** [comment]
@azerothcore/testers

**Review by Helias — APPROVED** [review]

**Comment by pangolp** [comment]
36 queries executed, 36 success, 0 errors, 0 warnings

The queries run smoothly on the latest version of the emulator. https://github.com/azerothcore/azerothcore-wotlk/commit/d72375c4ebe8ba132ec81ec87baeb71c3cf13ff7

**Comment by Sedatyf** [comment]
Thank you pangolp, but you have to test it in game as well. Please see my test on other Outlands area to have a rough idea about how to do it

**Comment by pangolp** [comment]
> Thank you pangolp, but you have to test it in game as well. Please see my test on other Outlands area to have a rough idea about how to do it

Yes, I read a comment on another pull request that mentioned that. But at the moment, I don't have enough time to be able to do this kind of testing. I'll keep looking for shorter PRs.

**Comment by Sedatyf** [comment]
Well, you can do split your test in several little test and post results when you're done. That's why I'm doing on one area. Other tests have been more urgent than those ones, so I'll send my results at the end

**Comment by wetbrownsauce** [comment]
Beginning to test this one. I have evaluated all of the PRE-PR behavior, and as expected, all addressed mobs do nothing but melee, with the exception of the [Skettis Wing Guard](https://wowgaming.altervista.org/aowow/?npc=21644) which appears to function as intended already.  

I will apply the SQL in a little bit and test them all POST-PR, as well as hop into burning crusade classic and compare to the mobs in there. But I have to shower and do some things first. 

[Skettis Wing Guard](https://wowgaming.altervista.org/aowow/?npc=21644) appears to function almost perfectly already, and the SmartAI listed on aowow does not match the script added through this PR. Hence why I would like to test on live servers to compare.  

Also, is there a reason why the flee for help scripts are always 5-30% hp? If a player in theory brought a warped peon down from 50% hp to 1% hp in one hit, should it not still flee? 

Stay tuned for POST-PR testing and live server testing.

**Comment by wetbrownsauce** [comment]
Alright so I finally had time to sit down and test and compare. Here are my results followed at the end by my thoughts:

### [Shadowwing Owl](https://wowgaming.altervista.org/aowow/?npc=22265)
Pre-PR:
- Only uses basic melee.

Post-PR:
- Uses screech between 20%-80%. Only does this once.

Live (TBC-Classic):
- Uses [Eagle Claw](https://wowgaming.altervista.org/aowow/?spell=30285) in combat occasionally.
- Does NOT use screech.

---

### [Shienor Talonite](https://wowgaming.altervista.org/aowow/?npc=18449)
Pre-PR:
- Only uses basic melee.

Post-PR:
- Appears to use dual wield.
- Uses Thrash.

Live (TBC-Classic):
- Appears to use dual wield. 
- Does NOT appear to use Thrash. 

---

### [Tuurem Scavenger](https://wowgaming.altervista.org/aowow/?npc=18456)
Pre-PR:
- Only uses basic melee.
- Does not flee when low hp.

Post-PR:
- Uses Thrash.
- Flees when hp between 5%-30%.

Live (TBC-Classic):
- Does NOT appear to use Thrash. 
- Does NOT flee when low hp.

---

### [Firewing Courier](https://wowgaming.altervista.org/aowow/?npc=18548)
Pre-PR:
- Only uses basic melee.

Post-PR:
- Casts Sun Shield on aggro. (this mob has no mana and sun shield grants mana regen so why would it do this?)

Live (TBC-Classic):
- Only uses basic melee. 
- Does NOT use Sun Shield.

---

### [Vengeful Husk](https://wowgaming.altervista.org/aowow/?npc=22045)
Pre-PR:
- Only uses basic melee.

Post-PR:
- Uses Debilitating Strike during combat.

Live (TBC-Classic):
- Uses Debilitating Strike during combat.

---

### [Vengeful Draenei](https://wowgaming.altervista.org/aowow/?npc=21636)
Pre-PR:
- Only uses basic melee.

Post-PR:
- Uses Thrash in combat.

Live (TBC-Classic):
- Does not exists at location. Couldn't test. 

---

### [Warped Peon](https://wowgaming.altervista.org/aowow/?npc=18595)
Pre-PR:
- Does not flee when low hp.

Post-PR:
- Flees when 5%-30% hp.

Live (TBC-Classic):
- Flees when low hp.

---

### [Time-Lost Skettis High Priest](https://wowgaming.altervista.org/aowow/?npc=21787)
Pre-PR:
- Only uses basic melee.

Post-PR:
- Casts Shadowform on spawn.
- Casts Shadow Word: Pain on aggro. Only once.
- Casts Holy Smite during combat.
- Uses Flash Heal when 10%-30% hp. Only once.

Live (TBC-Classic):
- Not in game yet. Comes in a later phase. Unable to test. 

---

### [Skettis Wing Guard](https://wowgaming.altervista.org/aowow/?npc=21644)
Pre-PR:
- Uses Demoralizing Shout in combat. First cast after 1-3sec, repeats every 10ish sec.
- Uses Gushing Wound in combat. First cast after 4-7sec, repeats every 30ish sec when debuff expires.
- Uses retaliate in combat. First cast after 6-9sec, repeats every 11ish sec.
- Enrages when below 30% hp, and emotes in chat that it is enraged.

Post-PR:
- Uses Demoralizing Shout after 1sec, repeats every 30sec. 
- Uses Gushing Wound after 5ish sec, repeats every 30 seconds.
- Uses Retaliate ONCE when between 20%-80% hp.
- Enrages between 5%-30% hp and emotes in chat that it is enraged.

Live (TBC-Classic):
- Not in game yet. Comes in a later phase. Unable to test.

---

### [Skettis Sentinel](https://wowgaming.altervista.org/aowow/?npc=21912)
Pre-PR:
- Only uses basic melee.

Post-PR:
- Uses Debilitating Strike during combat.
- Uses Cleave ONCE between 20%-80% hp.
- Enrages at 5%-30% hp.

Live (TBC-Classic):
- Not in game yet. Comes in a later phase. Unable to test.

---

### [Skettis Time-Shifter](https://wowgaming.altervista.org/aowow/?npc=21652)
Pre-PR:
- Only uses basic melee.

Post-PR:
- Uses Shriveling Gaze on aggro. Only once.
- Uses Chain Lightning during combat. 

Live (TBC-Classic):
- Not in game yet. Comes in a later phase. Unable to test. 

---

### [Time-Lost Skettis Reaver](https://wowgaming.altervista.org/aowow/?npc=21651)
Pre-PR:
- Only uses basic melee.

Post-PR:
- Uses Battle Shout on aggro. Only once.
- Uses Cleave during combat. 
- Uses Frenzy at 5%-30% hp.

Live (TBC-Classic):
- Not in game yet. Comes in a later phase. Unable to test. 

---

### Thoughts
Everything I am about to say assumes that these mobs did not change their abilities/behaviors from TBC to WotLK. If someone has evidence to the contrary, please provide it. My comparisons above hopefully outline enough that if the mobs changed between TBC and WotLK, it will be easily verified whether or not this PR matches up with that.  

Warped Peon and Vengeful Husk (and maybe some skettis mobs) appear to function correctly POST-PR. If they were the only two here, I would say it works perfectly. However, many do not match the behavior on live servers, or just don't seem correct. The Shadowwing Owl on live servers uses a completely different ability than what it present here, and the Firewing Courier on live servers does not use the spell included in this PR. Further, the spell that it does use in this PR is one which affects mana regen, of which the Courier does not even have (exactly as it is on live). The Vengeful Draenei is not at the location on live servers that it is on here, so I cannot find and test it. Live servers do not yet have Skettis mobs/quests, as those came later in TBC. So I cannot even test those. 

However, [aowow has SmartAI listed for Skettis Wing Guard](https://wowgaming.altervista.org/aowow/?npc=21644), and the AI listed there matches what already happens in AzerothCore PRE-PR. The POST-PR changes actually make this mob behave quite differently and I am unsure if that is correct.   

Lastly, does anyone know why we are using 5%-30% hp to mark when to flee for help or enrage? If I bring a mob down from 50% hp to 1% hp in a single hit, it does neither. I don't think that is correct but lack the precision to execute such a specific amount of damage on live servers.

**Comment by steenburgh** [comment]
Nice super detailed test report, @wetbrownsauce!

**Comment by necropola** [comment]
@T1ti Do you/we have any sniffed data (from vmagos) for tbc.classic creatures?

**Comment by FrancescoBorzi** [comment]
> @wetbrownsauce AFAIK [aowow](https://wowgaming.altervista.org/aowow/) uses AC World DB data as its source.

yes, correct. Not up to date

**Comment by Knindzagxg** [comment]
Note to my self to do checking:

Shadowwing Owl - Claw instead Screech.
Firewing Courier - Check Sun Shield.
Tuurem Scavenger - Trash and Flee.
Shienor Talonite - Thrash.

About Skettis NPC's - Only reason why Guard was changed is due crazy spell queues.
If i remember he was kinda spamming Retaliation which made him almost unkillable.
And same for Demo Shout, since there's no reason to use it every 9 seconds since debuff is 30 sec.

Main prob that we can't find correct timers unless it's a boss sadly.
Will try to get more info about it, but tricky.

**Comment by Knindzagxg** [comment]
Shadowwing Owl - Checked but there is no claw. Claw is only used when this NPC is tamed (Also require Focus to use it, which only hunter pets have) Removed Screech tho cause it's added in later expansions.

Firewing Courier - Removed SAI -> Added later as well.

Tuurem Scavenger - Same as Courier - Removed SAI.

Shienor Talonite - Thrash attack removed.

All sorted now xP
Thanks for testing and sorry cause late update on this xP

**Comment by Knindzagxg** [comment]
Shadowwing Owl updated.

**Comment by wetbrownsauce** [comment]
@Knindzagxg 
Let me know when this PR is ready to be re-tested so we can get it through faster!

**Comment by Knindzagxg** [comment]
Updated already those u mentioned.

Everyone should be good now -> You can retest.

**Comment by Knindzagxg** [comment]
Oh btw, forgot to answer on your question about:

"Lastly, does anyone know why we are using 5%-30% hp to mark when to flee for help or enrage? If I bring a mob down from 50% hp to 1% hp in a single hit, it does neither. I don't think that is correct but lack the precision to execute such a specific amount of damage on live servers."

Main reason why is inside your question xP
Problem is that calculation when things happen at %hp. Thats why on Blizz we have creatures flee for assist or enrage bellow 30%.
In some cases, flee can be even at lower %. And like you said, if you bring mob from 50 to 1% in a hit, sometimes it wont work for unknown reason. Thats why there is a gap for event to happen and its gonna happen most of times during leveling.
Later on, pretty much everyone will 1 shoot those mobs, but its pretty important for new characters during leveling, since none will have end game item level or etc.

**Comment by Knindzagxg** [comment]
Need checking of those few changes that were requested, so i can move to Northrend Zones.

**Comment by Knindzagxg** [comment]
Retested required changes -> Everything working now as suggested.

**Pull request merged**